internal/querynodev2/delegator/delegator_data.go

```diff
@@ -749,7 +749,7 @@ func (sd *shardDelegator) readDeleteFromMsgstream(ctx context.Context, position
 			for _, tsMsg := range msgPack.Msgs {
 				if tsMsg.Type() == commonpb.MsgType_Delete {
 					dmsg := tsMsg.(*msgstream.DeleteMsg)
-					if dmsg.CollectionID != sd.collectionID || dmsg.GetPartitionID() != candidate.Partition() {
+					if dmsg.CollectionID != sd.collectionID || (dmsg.GetPartitionID() != common.AllPartitionsID && dmsg.GetPartitionID() != candidate.Partition()) {
```

Comment:
i'm sure there is a util function to check whether the partition matching.
this has happened many times, and every new one will miss to handle the magic `AllPartitionsID` before they know it exists, it's worth effort to make something like `optional<T>`

Reply:
btw this line is too long

Reply:
- `&&` is with higher priority so the braces are not needed
- this line is too long

Reply:
What `optional<T>` means?
It is certain that we should do something to make sure `AllPartitionsID` was known when someone want to verify partitionID, but these maybe next pr.

Reply:
> i'm sure there is a util function to check whether the partition matching. this has happened many times, and every new one will miss to handle the magic `AllPartitionsID` before they know it exists, it's worth effort to make something like `optional<T>`

What optional<T> means?
It is certain that we should do something to make sure AllPartitionsID was known when someone want to verify partitionID, but these maybe next pr.

Reply:
@yah01 do you mean optional partition id for delete operation here? Same problem still persist that all usage shall check partition id is specified since there is no `Optional<T>` implemented yet in golang
@aoiasd we could extract the check logic into utility function in next pr
merging patch first